src/pytest_run_parallel/plugin.py

```diff
@@ -97,6 +100,47 @@ def closure(*args, **kwargs):
     return inner
 
 
+def identify_warnings_handling(fn):
+    src = inspect.getsource(fn)
+    tree = ast.parse(dedent(src))
+    catches_warns = False
+    blacklist = {
+        ("pytest", "warns"),
+        ("pytest", "deprecated_call"),
+        ("warnings", "catch_warnings"),
+    }
+    modules = {mod for mod, _ in blacklist}
+    modules_aliases = {}
+    func_aliases = {}
+    for var_name in fn.__globals__:
+        value = fn.__globals__[var_name]
+        if inspect.ismodule(value):
+            if value.__name__ in modules:
+                modules_aliases[var_name] = value.__name__
+        elif inspect.isfunction(value):
+            real_name = value.__name__
+            for mod in modules:
+                if mod in value.__module__:
+                    func_aliases[var_name] = (mod, real_name)
```

Comment:
Here, we can break afterwards to avoid doing extra iterations that we know will fail.

```suggestion
                    func_aliases[var_name] = (mod, real_name)
                    break
```